src/utils/details.js

```diff
@@ -0,0 +1,196 @@
+'use strict'
+
+const parseHeaders = require('parse-headers')
+const base64url = require('./base64url')
+const cryptoHelper = require('./crypto')
+const Packet = require('./packet')
+const debug = require('debug')('ilp:psk-data')
+const { startsWith } = require('.')
+const DATA_DELIMITER = '\n\n'
+const STATUS_LINE_REGEX = /^PSK\/1\.0$/
+const KEY_HEADER_REGEX = /^hmac-sha-256 (.+)$/
+
+function _createRequest ({
+  statusLine,
+  headers,
+  data
+}) {
+  const statusLineText = statusLine ? 'PSK/1.0\n' : ''
+  const headerLines = Object.keys(headers)
+    .map((k) => k + ': ' + headers[k])
+    .join('\n') + DATA_DELIMITER
+
+  let rawData = data
+  if (!Buffer.isBuffer(data) && typeof data === 'object') {
```

Comment:
I might have originally argued for this behavior but I think it would be better for the transport layer to expect a buffer and reject/throw if you pass anything else. The SPSP layer should convert the JSON to a buffer